ZeeguuAPI/ZeeguuAPI/ZeeguuAPI.swift

```diff
@@ -0,0 +1,334 @@
+	///
+	/// - returns: `true` if a user is logged in, `false` otherwise.
+	public var isLoggedIn: Bool {
+		get {
+			return currentSessionID != 0
+		}
+	}
+	
+	/// Get the `ZeeguuAPI` instance. This method is the only way to get an instance of the ZeeguuAPI class.
+	///
+	/// - returns: The shared `ZeeguuAPI` instance.
+	public static func sharedAPI() -> ZeeguuAPI {
+		return instance;
+	}
+	
+	private init() {
+		self.currentSessionID = 0
+		let def = NSUserDefaults.standardUserDefaults()
+		if (def.objectForKey(ZeeguuAPI.sessionIDKey) != nil) {
+			self.currentSessionID = def.objectForKey(ZeeguuAPI.sessionIDKey)!.integerValue
+		}
+	}
+	
+	/// Registers a user.
+	///
+	/// After the user is registered and logged in, you can use the ZeeguuAPI object to make requests on behalf of the user.
+	///
+	/// - parameter username: The username of the user to register.
+	/// - parameter email: The email address of the user.
+	/// - parameter password: The password of the user.
+	/// - parameter completion: A block that receives a success parameter, which is true if the user was logged in successfully.
+	public func registerUserWithUsername(username: String, email: String, password: String, completion: (success: Bool) -> Void) {
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.AddUser, pathComponents: [email], method: HTTPMethod.POST, parameters: ["username":username, "password":password])
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil) {
+				let sesID = Int(response!)
+				if (sesID != nil) {
+					self.currentSessionID = sesID!
+				}
+				completion(success: true)
+			} else {
+				completion(success: false)
+			}
+		}
+	}
+	
+	/// Logs a user in.
+	///
+	/// After the user is logged in, you can use the ZeeguuAPI object to make requests on behalf of the user.
+	/// 
+	/// - parameter email: The email address of the user to log in.
+	/// - parameter password: The password of the user.
+	/// - parameter completion: A block that receives a success parameter, which is true if the user was logged in successfully.
+	public func loginWithEmail(email: String, password: String, completion: (success: Bool) -> Void) {
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.Session, pathComponents: [email], method: HTTPMethod.POST, parameters: ["password":password])
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil) {
+				let sesID = Int(response!)
+				if (sesID != nil) {
+					self.currentSessionID = sesID!
+				}
+				completion(success: true)
+			} else {
+				completion(success: false)
+			}
+		}
+	}
+	
+	/// Returns the language code of the learned langugage of the logged in user.
+	///
+	/// - parameter completion: A block that will receive a language code of the learned language or nil if the request couldn't be completed.
+	public func learnedLanguage(completion: (langCode: String?) -> Void) {
```

Comment:
You use "set" in front of the method names who are doing POST. 
It would be cool to use "get" in front of those methods that are going GET.